algorithms/src/merkle_tree/merkle_tree.rs

```diff
@@ -296,14 +296,28 @@ impl<P: MerkleParameters + Send + Sync> MerkleTree<P> {
     fn hash_row<L: ToBytes + Send + Sync>(
         parameters: &P,
         leaves: &[L],
-    ) -> Result<Vec<Vec<<<P as MerkleParameters>::H as CRH>::Output>>, MerkleError> {
+    ) -> Result<Vec<Vec<<<P as MerkleParameters>::LeafCRH as CRH>::Output>>, MerkleError> {
         match leaves.len() {
             0 => Ok(vec![]),
             _ => {
                 Ok(vec![crate::cfg_iter!(leaves).map(|leaf| parameters.hash_leaf(&leaf).unwrap()).collect::<Vec<_>>()])
             }
         }
     }
+
+    fn hash_row_inner<L: ToBytes + Send + Sync>(
+        parameters: &P,
+        inner: &[L],
+    ) -> Result<Vec<Vec<<<P as MerkleParameters>::TwoToOneCRH as CRH>::Output>>, MerkleError> {
+        match inner.len() {
+            0 => Ok(vec![]),
+            _ => Ok(vec![
+                crate::cfg_iter!(inner)
+                    .map(|inner_nodes| parameters.hash_two_to_one(&inner_nodes).unwrap())
```

Comment:
since the result is a `Result<Vec<T, MerkleError>>`, perhaps we can avoid the calls to `unwrap` here and remove the enclosing `Ok()` to improve this method's safety?

Reply:
Agreed, I've removed the `unwraps` in [cab5f14](https://github.com/AleoHQ/snarkVM/pull/753/commits/cab5f149842eadd899ee2285e4c48980b74f2f75).

I also noticed that we unnecessarily had a double Vec that was seemingly unused due to our `hash_leaf_row` and `hash_two_to_one_row`. I've updated the output to `Result<Vec<T, MerkleError>>` from `Result<Vec<Vec<T, MerkleError>>>` and made the corresponding changes above.